### PR TITLE
Add CI and test skeleton

### PR DESCRIPTION
**Pull request by ml-evs**
Hey @MJCliffe, here's a quick hash of how testing could work.

I haven't been able to test locally as the WiFi here isn't good enough to download the requirements before I'm due to leave :upside_down_face: 

Only issues I can foresee: the tests need to import the app code. In the current layout of the source code this might be a bit tricky, but there's normally a hacky way around it.

For the CI tests I was imagining that we just push the sample data through the app, make sure it doesn't crash, and then do some light analysis on the returned HTML. I don't think it would be very useful to you to have tests that you would need to update every time you add something new to the server, but hopefully some lightweight tests will prevent you from messing up the live server.

### CODE REVIEW

**Comment by ml-evs** [comment]
(You might need to enable GitHub actions in the settings for this repo -- can't remember the exact workflow for it -- I would say this PR is ready once those remote tests start passing).

**Comment by ml-evs** [comment]
@MJCliffe this should be good to go -- it works locally but will need to test in the CI, I think you need to approve the workflow somewhere on this page before it will run?

**Review by MJCliffe — APPROVED** [review]

**Comment by MJCliffe** [comment]
There's an error in the dependencies somehow: I'm afraid I don't know what the best way to fix this.

`ERROR: Directory '.' is not installable. Neither 'setup.py' nor 'pyproject.toml' found.`

**Comment by ml-evs** [comment]
Ah that's my bad, left some boilerplate in for installing the package in the repo, but there isn't one for this repo. If I delete that line it should work.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        cache-dependency-path: |
+          requirements*.txt
+
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -U setuptools wheel
+        pip install -r requirements.txt -r requirements-test.txt
+        pip install .
```
Comment (ml-evs) [inline .github/workflows/ci.yml]:
```suggestion
```

**Comment by ml-evs** [comment]
Unfortunately you need to reapprove the tests running again, but that should fix that issue at least!

**Pull request merged**